src/commands/utils_command/ledger_publickey_subcommand/mod.rs

```diff
@@ -0,0 +1,63 @@
+use std::str::FromStr;
+
+/// Generate a key pair of secret and public keys (use it anywhere you need
+/// Ed25519 keys)
+#[derive(Debug, clap::Clap, Clone)]
+pub struct CliLedgerPublicKey {
+    #[clap(long, default_value = "44'/397'/0'/0'/1'")]
+    pub seed_phrase_hd_path: slip10::BIP32Path,
+    #[clap(long, default_value = "plaintext")]
+    pub format: crate::common::OutputFormat,
+}
+
+impl Default for CliLedgerPublicKey {
+    fn default() -> Self {
+        Self {
+            seed_phrase_hd_path: slip10::BIP32Path::from_str("44'/397'/0'/0'/1'").unwrap(),
+            format: crate::common::OutputFormat::Json,
+        }
+    }
+}
+
+impl CliLedgerPublicKey {
+    pub async fn process(self) -> crate::CliResult {
+        println!(
+            "Please allow getting the PublicKey on Ledger device (HD Path: {})",
+            crate::common::bip32path_to_string(&self.seed_phrase_hd_path)
+        );
+        let public_key = match near_ledger::get_public_key(self.seed_phrase_hd_path.clone()).await {
+            Ok(public_key) => public_key,
+            Err(near_ledger_error) => {
+                println!(
+                    "An error occurred while trying to get PublicKey from Ledger device: {:?}",
+                    near_ledger_error
+                );
+                return Ok(());
+            }
```

Comment:
Let's return `Err` with the color_eyer Report like here:
 
https://github.com/FroVolod/near-cli/blob/master/src/commands/construct_transaction_command/sign_transaction/sign_with_private_key/mod.rs#L123-L126